oh-my-zsh.sh

```diff
@@ -59,9 +59,54 @@ if [ -z "$ZSH_COMPDUMP" ]; then
   ZSH_COMPDUMP="${ZDOTDIR:-${HOME}}/.zcompdump-${SHORT_HOST}-${ZSH_VERSION}"
 fi
 
-# Load and run compinit
-autoload -U compinit
-compinit -i -d "${ZSH_COMPDUMP}"
+# If one or more completion-dependent directories have insecure ownership or
+# permissions, the low-level compaudit() function fails with output resembling:
+#
+#     There are insecure directories:
+#     /usr/share/zsh/site-functions
+#     /usr/share/zsh/5.0.6/functions
+#     /usr/share/zsh
+#     /usr/share/zsh/5.0.6
+#
+# Silently detect and, in a few well-known cases, offer to fix this issue.
+if ! compaudit &>/dev/null; then
+  # Print this list of insecure directories.
+  compaudit
+  print
+
+  # If this is Cygwin *and* both standard input and output are connected to an
+  # interactive terminal, offer to fix this. Cygwin has a long-standing bug in
```

Comment:
This all looks awesome!

One little thing: could you please find a reference to that _"long-standing bug"_ and include it in the comment?

I found [this](http://www.wezm.net/technical/2008/09/zsh-cygwin-and-insecure-directories/) but I am more looking for something about Cygwin messing up permission on those directories.